hs-abci-sdk/src/Tendermint/SDK/Modules/Validators/Keeper.hs

```diff
@@ -57,17 +57,18 @@ getQueuedUpdatesF
   :: Members [ReadStore, Error AppError] r
   => Sem r (Map.Map PubKey_ Word64)
 getQueuedUpdatesF = L.foldl (\m (ValidatorUpdate_ ValidatorUpdate{..}) ->
-  Map.alter (Just . fromMaybe (toWord validatorUpdatePower)) (PubKey_ validatorUpdatePubKey) m) Map.empty updatesList
+  Map.alter (Just . fromMaybe (toWord validatorUpdatePower)) (toPK_ validatorUpdatePubKey) m) Map.empty updatesList
   where
     toWord (WrappedVal x) = fromIntegral x
+    toPK_ = PubKey_ . fromJust
```

Comment:
Sorry, one last thing. We prefer not to use partial functions like this, its better to use things like `error` to at least print a message that can help debug

Reply:
So like `fromMaybe (error “my error message”`

Reply:
Fair enough. This should never happen, because it's only unwrapping the Just that was put on by `queueUpdateF`, but since it's not represented in the types it's reasonable to be total / use an error message. Will do :+1:

Reply:
Yeah it's just one of those things that would be nearly impossible to track down, especially since the assumption is so strong.